util/design/BUILD

```diff
@@ -22,7 +22,11 @@ py_library(
 py_binary(
     name = "gen-flash-img",
     srcs = ["gen-flash-img.py"],
-    deps = [":secded_gen"],
+    deps = [
+        ":secded_gen",
+        requirement("galois"),
+        requirement("numpy"),
```

Comment:
Is the `numpy` dependency really needed? Or are there alternatives? Because this caused issues in the past. IIRC `numpy` itself is just rather big and has other dependencies. This may cause just the setup phase in CI jobs to take quite a lot more time already.

More than once we had to work around this. I think in one case we ended up installing a tarball (IIRC for ChipWhisperer). In the meantime they luckily released a minimal version of the API without the critical `numpy/scipy` dependencies.

Reply:
Unfortunately, `galois` requires `numpy`. Yes, I am aware of the issues, another being we had compatibility issues with python version between Ubuntu 20.04 vs 18.04 vs what is managed by Bazel.

I originally tried [`pyfinite`](https://github.com/emin63/pyfinite), but encountered some issues with the old polynomial. Let me try it again with the new polynomial in #17443, and get back to you.

Reply:
I resolved the issues and we now use `pyfinite` instead of `numpy`. Thanks @vogelpi !

Reply:
Many thanks for taking another look at this. Good that we managed to avoid `numpy` once again :-)